### PR TITLE
Add BBEs for the LDAP module

### PR DESCRIPTION
**Pull request by Nuvindu**

### CODE REVIEW

**Review by DimuthuMadushan — COMMENTED** [review]
LGTM!

**Review by DimuthuMadushan — APPROVED** [review]

File: examples/ldap-add-remove-entry/ldap_add_remove_entry.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+The `add` operation creates a new entry in a directory server. It requires a `DN` (Distinguished Name) for the entry and the attributes to be included. The `objectClass` attribute must be specified to define the object classes for the entry, and any attributes required by these object classes should also be included.
+
+The `delete` operation removes an entry from a directory server. It requires a `DN` (Distinguished Name) of the entry to be deleted.
+
+::: code ldap_add_remove_entry.bal :::
+
+## Prerequisites
+
+- Make sure to keep an LDAP server up and running while running the example.
```
Comment (NipunaRanasinghe) [inline examples/ldap-add-remove-entry/ldap_add_remove_entry.md:13]:
```suggestion
- Ensure that an LDAP server is up and running locally on port 389 while running the example.
```

File: examples/ldap-search-entry/ldap_search_entry.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+- `baseDN`: Defines the base of the subtree for the search.
+- `filter`: Specifies the criteria that entries must meet.
+- `scope`: Determines extent of the the search coverage within the subtree.
+
+When a directory server processes a valid search request, it finds entries within the defined scope that meet the filter criteria. The matching entries are then returned to the client in a `ldap:SearchResult` record.
+
+::: code ldap_search_entry.bal :::
+
+## Prerequisites
+- Make sure to keep an LDAP server up and running while running the example.
```
Comment (NipunaRanasinghe) [inline examples/ldap-search-entry/ldap_search_entry.md:16]: ditto

File: examples/ldap-add-remove-entry/ldap_add_remove_entry.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+- Run the example by executing the command below.
+
+::: out ldap_add_remove_entry.server.out :::
+
+## Related links
+
+- [`ldap:Client` - API documentation](https://lib.ballerina.io/ballerina/ldap/latest#Client)
+- [`ldap` module - API documentation](https://lib.ballerina.io/ballerina/ldap/latest/)
```
Comment (ayeshLK) [inline examples/ldap-add-remove-entry/ldap_add_remove_entry.md:22]: Rather this lets point to the API spec of the `ldap:Client`

File: examples/ldap-search-entry/ldap_search_entry.bal [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import ballerina/io;
+import ballerina/ldap;
+
+public function main() returns error? {
+    // Initializes a new LDAP client.
+    ldap:Client ldapClient = check new ({
```
Comment (ayeshLK) [inline examples/ldap-search-entry/ldap_search_entry.bal:6]: Does LDAP client `init` supports included record params (IMO it should) ? If that is the case lets use that approach here. That is the preferred way when we init clients.

File: examples/ldap-search-entry/ldap_search_entry.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+- Run the example by executing the command below.
+
+::: out ldap_search_entry.server.out :::
+
+## Related links
+- [`ldap:Client` - API documentation](https://lib.ballerina.io/ballerina/ldap/latest#Client)
+- [`ldap` module - API documentation](https://lib.ballerina.io/ballerina/ldap/latest/)
```
Comment (ayeshLK) [inline examples/ldap-search-entry/ldap_search_entry.md:24]: ditto

File: examples/ldap-add-remove-entry/ldap_add_remove_entry.bal [hunk 6]
```diff
@@ -0,0 +1,27 @@
+import ballerina/io;
+import ballerina/ldap;
+
+public function main() returns error? {
+    // Initializes a new LDAP client with credentials.
+    ldap:Client ldapClient = check new ({
```
Comment (ayeshLK) [inline examples/ldap-add-remove-entry/ldap_add_remove_entry.bal:6]: Refer to this comment [1]

[1] - https://github.com/ballerina-platform/ballerina-distribution/pull/5578/files#r1718020943

File: examples/ldap-search-entry/ldap_search_entry.md (hunk 5 above)
Comment (Nuvindu) [reply examples/ldap-search-entry/ldap_search_entry.md:24]: Fixed in [b30e9aa](https://github.com/ballerina-platform/ballerina-distribution/pull/5578/commits/b30e9aa47ee2c5298c77d8a40a9666cc45004f01)

File: examples/ldap-add-remove-entry/ldap_add_remove_entry.md (hunk 3 above)
Comment (Nuvindu) [reply examples/ldap-add-remove-entry/ldap_add_remove_entry.md:20]: Fixed in [b30e9aa](https://github.com/ballerina-platform/ballerina-distribution/pull/5578/commits/b30e9aa47ee2c5298c77d8a40a9666cc45004f01)

File: examples/ldap-add-remove-entry/ldap_add_remove_entry.md [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+## Prerequisites
+- Ensure that an LDAP server is up and running locally on port 389 while running the example.
+
+- Run the example by executing the command below.
+
+::: out ldap_add_remove_entry.server.out :::
+
+## Related links
+- [`ldap:Client` - Specification](/spec/ldap/#2-ldap-client)
+- [`ldap` module - API documentation](https://lib.ballerina.io/ballerina/ldap/latest/)
```
Comment (ayeshLK) [inline examples/ldap-add-remove-entry/ldap_add_remove_entry.md:20]: Lets re-order this:
1. LDAP client `add` API documentation
2. LDAP client `delete` API documentation
3. LDAP Client API specification

File: examples/ldap-search-entry/ldap_search_entry.md [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+## Prerequisites
+- Ensure that an LDAP server is up and running locally on port 389 while running the example.
+
+- Run the example by executing the command below.
+
+::: out ldap_search_entry.server.out :::
+
+## Related links
+- [`ldap:Client` - Specification](/spec/ldap/#2-ldap-client)
+- [`ldap` module - API documentation](https://lib.ballerina.io/ballerina/ldap/latest/)
```
Comment (ayeshLK) [inline examples/ldap-search-entry/ldap_search_entry.md:24]: ditto

**Review by ayeshLK — APPROVED** [review]
LGTM

**Pull request merged**